rust/connlib/tunnel/src/utils.rs

```diff
@@ -35,7 +37,36 @@ pub fn turn(
                 None
             }
         })
-        .filter(|(_, socket, _, _, _)| predicate(socket))
+        .group_by(|(id, _, _, _, _)| *id)
+        .into_iter()
+        .filter_map(|(_, grouped)| {
+            grouped.reduce(
+                |(_, current_socket, _, _, _), (id, socket, username, password, realm)| {
+                    let new_socket = match (current_socket, socket) {
+                        (RelaySocket::V4(v4), RelaySocket::V6(v6)) => RelaySocket::Dual { v4, v6 },
+                        (RelaySocket::V6(v6), RelaySocket::V4(v4)) => RelaySocket::Dual { v4, v6 },
+                        (_, dual @ RelaySocket::Dual { .. })
+                        | (dual @ RelaySocket::Dual { .. }, _) => {
+                            tracing::warn!(%id, "Duplicate addresses for relay");
```

Comment:
I can add one. I am not sure how useful it is. We almost never run a debug build where those would be present and I think we shouldn't upgrade this to a normal `assert!` because there is no reason to fail here as connlib can still function normally.

The warning on the other hand will be visible always and still allows us to investigate. We should see it on the gateway too which we will hopefully catch once we have the soak tests.